addons/account_edi_proxy_client/models/account_edi_proxy_user.py

```diff
@@ -64,10 +65,25 @@ class AccountEdiProxyClientUser(models.Model):
 
     _sql_constraints = [
         ('unique_id_client', 'unique(id_client)', 'This id_client is already used on another user.'),
-        ('unique_edi_identification', 'unique(edi_identification, proxy_type, edi_mode)', 'This edi identification is already assigned to a user'),
-        ('unique_company_proxy', 'unique(company_id, proxy_type, edi_mode)', 'This company has a user already created for this EDI type'),
+        ('unique_active_edi_identification', '', 'This edi identification is already assigned to an active user'),
+        ('unique_active_company_proxy', '', 'This company has an active user already created for this EDI type'),
     ]
 
+    def _auto_init(self):
+        super()._auto_init()
+        if not index_exists(self.env.cr, 'account_edi_proxy_client_user_unique_active_edi_identification'):
+            self.env.cr.execute("""
+                CREATE UNIQUE INDEX account_edi_proxy_client_user_unique_active_edi_identification
+                                 ON account_edi_proxy_client_user(edi_identification, proxy_type, edi_mode)
+                              WHERE (active = True)
+            """)
+        if not index_exists(self.env.cr, 'account_edi_proxy_client_user_unique_active_company_proxy'):
+            self.env.cr.execute("""
+                CREATE UNIQUE INDEX account_edi_proxy_client_user_unique_active_company_proxy
+                                 ON account_edi_proxy_client_user(company_id, proxy_type, edi_mode)
+                              WHERE (active = True)
+            """)
```

Comment:
I don't get why you don't keep the previous code here.
`('unique_active_edi_identification', 'UNIQUE(edi_identification, proxy_type, edi_mode)', 'This edi identification is already assigned to an active user'),`

Reply:
because if a peppol user migrates away from odoo, their user is archived but still there.
If they decide to register again, they won't be able to, because an edi user with their `edi_identification` already exists, even if it's archived.
So, we want to apply this constraint only to active users that are not archived, hence the necessary condition `active = True`

Reply:
@aliyatastemirova Ok got it now